components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -831,23 +844,18 @@ private void deleteInbound(String applicationId, String inboundType) {
         // Delete the associated CORS origins if the inboundType is oauth2.
         if (inboundType.equals(OAUTH2)) {
             String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
-            int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
-            CORSOriginDAO corsOriginDAO = new CORSOriginDAOImpl();
+            CORSManagementService corsManagementService = ApplicationManagementServiceHolder.getCorsManagementService();
             List<CORSOrigin> existingCORSOrigins = null;
-            ApplicationBasicInfo applicationBasicInfo = null;
             try {
-                applicationBasicInfo = ApplicationManagementService.getInstance()
-                        .getApplicationBasicInfoByResourceId(applicationId, tenantDomain);
-                existingCORSOrigins = corsOriginDAO.getCORSOriginsByApplicationId(
-                        applicationBasicInfo.getApplicationId(), tenantId);
-                corsOriginDAO.deleteCORSOrigins(applicationBasicInfo.getApplicationId(), existingCORSOrigins.stream()
-                        .map(CORSOrigin::getId).collect(Collectors.toList()), tenantId);
-            } catch (CORSManagementServiceServerException | IdentityApplicationManagementException e) {
-                if (applicationBasicInfo != null && existingCORSOrigins != null) {
+                existingCORSOrigins = corsManagementService.getApplicationCORSOrigins(applicationId, tenantDomain);
+                corsManagementService.deleteCORSOrigins(applicationId, existingCORSOrigins.stream()
+                        .map(CORSOrigin::getId).collect(Collectors.toList()), tenantDomain);
+            } catch (CORSManagementServiceException e) {
+                if (existingCORSOrigins != null) {
                     try {
-                        corsOriginDAO.addCORSOrigins(applicationBasicInfo.getApplicationId(), existingCORSOrigins,
-                                tenantId);
-                    } catch (CORSManagementServiceServerException corsManagementServiceServerException) {
+                        corsManagementService.addCORSOrigins(applicationId, existingCORSOrigins.stream()
```

Comment:
why do we add them again? Is that due to another application having cors associations?

Reply:
It was to recover if the deletion fails in the middle. But it is handled in the dao layer to rollback. So removed this.